### ISSUE TITLE
--datadir is ignored during wallet and validator creation, etc

### ISSUE THREAD

**Issue by somuchid** [sigp/lighthouse#1571]
bug 1. --datadir has no effect during wallet creation, default folder .lighthouse is always used.

bug 2. --datadir is ignored during account creation, default folder .lighthouse is always used.

bug 3. validator process can only work normally on the default folder .lighthouse, error occurs when specifying a non-default --data-dir.

Error:

The validator_definitions.yml file does not contain either of the following fields for "/home/j/test/validators/0xaa8547c87726f0b936e29a510497400b8cb8be4af5cd99d53aa2f424e231b68b2fc78a2d0c3c1e055cb65cc84bc43d48/voting-keystore.json":

 - voting_keystore_password
 - voting_keystore_password_path

You may exit and update validator_definitions.yml or enter a password. If you choose to enter a password now then this prompt will be raised next time the validator is started.

Steps taken:
step 1. create a wallet in default folder
step 2. create a validator in default folder
step 3. rename folder .lighthouse to test
step 4. launch validator process against test folder, error
            command: lighthouse vc --datadir test
step 5. rename folder test back to .lighthouse
step 6. launch validator process with default folder, successfully create validator_definitions.yml
            command: lighthouse vc
step 7. rename folder .lighthouse to test
step 8. edit validator_definitions.yml and replace ".lighthouse" with "test"
step 9. launch validator process against test folder, error
            command: lighthouse vc --datadir test

Please make --datadir work. It is much needed. Both prysm and teku work fine with the switch.

**Comment by michaelsproul** [comment]
We're working on cleaning up our datadir handling in #1532 and will implement a fix for these issues shortly after that is merged. Thanks for the bug report

**Comment by michaelsproul** [comment]
If you want it to work in the meantime, you can use the different subflags for different directories.

For the wallet manager it is:

```
lighthouse account wallet --base-dir test/wallets
```

Similarly for the validator account manager it's:

```
lighthouse account validator --base-dir test/wallets
```

And for the validator client, it's (unfortunately) necessary to set the secrets dir separately:

```
lighthouse validator_client --datadir test --secrets-dir test/secrets
```

**Comment by paulhauner** [comment]
Resolved via #1532, but will only be included in the v0.3.0 release.
